draft-ietf-quic-http.md

```diff
@@ -633,18 +633,22 @@ tree could be discarded safely. Clients could potentially reference closed
 streams long after the server had discarded state, leading to disparate views of
 the prioritization the client had attempted to express.
 
-In HTTP/3, a number of placeholders are explicitly permitted by the server using
-the `SETTINGS_NUM_PLACEHOLDERS` setting. Because the server commits to
-maintaining these placeholders in the prioritization tree, clients can use them
-with confidence that the server will not have discarded the state. Clients MUST
-NOT send the `SETTINGS_NUM_PLACEHOLDERS` setting; receipt of this setting by a
-server MUST be treated as a connection error of type
+In HTTP/3, by using the `SETTINGS_NUM_PLACEHOLDERS` setting, the server
+advertises the number of client-controlled placeholders for which it is
+committing to maintain state.  Client-controlled placeholders are given their
+own number space that spans between zero and 2^62-1.  Clients can use the
+client-controlled placeholders with an ID less than the value of this setting
+with the confidence that the server will not have discarded the state.  The
```

Comment:
```suggestion
with the confidence that the server will discarded associated state.  The
```

Question, if we have PlaceholderN->RequestN->root in a tree and RequestN has long since been discarded; is this considered "discarding the state"?

Reply:
If the text is unclear, maybe we should change it to something like "with the confidence that the server will retain state"? Note that the sentence is about ID being *less* than SETTINGS_NUM_PLACEHOLDERS.

> Question, if we have PlaceholderN->RequestN->root in a tree and RequestN has long since been discarded; is this considered "discarding the state"?

No. "State" here means the existence of the placeholder and the relationship with it's children.

Reply:
Then say "that the server will not have disposed of the priority of the placeholder".

Reply:
Thank you for the suggestion. Applied in 47abc99 (changing "of the placeholder" to "of these placeholders").